src/IO/Observer/ObservationId.cpp

```diff
@@ -7,8 +7,34 @@
 #include <pup.h>
 
 namespace observers {
+ObservationKey::ObservationKey(const std::string& tag) noexcept
+    : key_(std::hash<std::string>{}(tag)) {}
+
+void ObservationKey::pup(PUP::er& p) noexcept { p | key_; }
+
+bool operator==(const ObservationKey& lhs, const ObservationKey& rhs) noexcept {
+  return lhs.key() == rhs.key();
+}
+
+bool operator!=(const ObservationKey& lhs, const ObservationKey& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+std::ostream& operator<<(std::ostream& os, const ObservationKey& t) noexcept {
+  return os << '(' << t.key() << ')';
```

Comment:
(not a change request, just something to consider improving) Since the `key` is the hash of the string, output is not very readable for the user. For example, this is the error message I got when debugging the linear solver observation:

```
Couldn't find registration key (11827073594539249116) in the registered observers.
```

If instead of the hash this printed the string, it would be more helpful. Of course this would mean storing the string instead of the hash, or both. What do you think?

Reply:
I agree. This is a small change so I'll make it.